### ISSUE TITLE
Remove duplicate trait bound or where clause

### ISSUE THREAD

**Issue by aldhsu** [rust-lang/rust-clippy#8674]
### What it does

This lint warns about repetitions of the same trait in trait bounds or where clauses.

### Lint Name

repetition_where_clause_or_trait_bound

### Category

pedantic

### Advantage

Leaving duplicate trait bounds is less readable.

### Drawbacks

None

### Example

```rust
pub fn foo<T: Copy + Copy>(t: T) {}
```
Could be written as:

```rust
pub fn foo<T: Copy>(t: T) {}
```

OR

```rust
pub fn foo<T>(t: T) where T: Copy + Copy {}
```
Could be written as:

```rust
pub fn foo<T>(t: T) where T: Copy {}
```

**Comment by aldhsu** [comment]
I saw there is an existing lint that compares where clauses and trait bounds for duplicates. This could be an extension of that lint.

**Comment by aldhsu** [comment]
Closed by #8703.
